duckduckgo/Android
# Cosmetic autoconsent rules

Task/Issue URL: https://app.asana.com/0/72649045549333/1203672644429320/f

**we should first release [Privacy Dashboard](https://github.com/duckduckgo/privacy-dashboard/pull/21) before merging this PR**

### Description

This adds support for the latest autoconsent version with cosmetic rules.

Related Privacy Dashboard PR: https://github.com/duckduckgo/privacy-dashboard/pull/21

### Steps to test this PR

1. enable the CPM feature
2. use VPN if you're not in Europe (cookie popups are rare in the US)
1. go to https://w3resource.com or https://bitcoin.org -> the animation should say "pop-up hidden" (cosmetic rule)
1. go to https://etsy.com or https://doodle.com/ -> the animation should say "pop-up managed"
2. Open Privacy Dashboard and click on the autoconsent-related row
3. on the next screen, click "Change in Settings" -> this should open the CPM settings screen
1. observe the changes in CPM feature description

### UI changes
### Animation
| Before  | After (cosmetic rule) | After (normal rule) |
| ------ | ----- | ---- |
| ![animation-managed](https://user-images.githubusercontent.com/2726132/213163738-653cd822-934a-4e1c-8fe6-f9f7e6a64925.jpg) | ![animation-hidden](https://user-images.githubusercontent.com/2726132/213163794-fbaea6b1-d466-48e0-9409-d3866bd0c588.jpg) | ![animation-managed](https://user-images.githubusercontent.com/2726132/213163738-653cd822-934a-4e1c-8fe6-f9f7e6a64925.jpg) |

### CPM settings
| Before  | After |
| ------ | ----- |
|![settings-before](https://user-images.githubusercontent.com/2726132/213165512-be2b8961-d839-4660-9e72-c44dc0347de3.jpg)|![settings-new](https://user-images.githubusercontent.com/2726132/213165546-fcc47882-5188-4ea9-ab20-b7136bb34c90.jpg)|

### Dashboard
| Before  | After (non-cosmetic, 1) | After (non-cosmetic, 2) |
| ------ | ----- | ----- |
| ![dashboard-before](https://user-images.githubusercontent.com/2726132/213232903-ff9f37fa-67a1-4642-91ea-0fd7cea1c419.jpg) | ![dashboard-after-1](https://user-images.githubusercontent.com/2726132/213233001-6345918c-5c61-4ff0-bf73-7b2e8f4ddf61.jpg) | ![dashboard-after-2](https://user-images.githubusercontent.com/2726132/213233061-6eda88df-115d-4b88-bb77-5c58ddc7e221.jpg) |

| Before  | After (cosmetic, 1) | After (cosmetic, 2) |
| ------ | ----- | ----- |
| ![dashboard-before](https://user-images.githubusercontent.com/2726132/213232903-ff9f37fa-67a1-4642-91ea-0fd7cea1c419.jpg) | ![dashboard-after-3](https://user-images.githubusercontent.com/2726132/213233684-1d9e0800-4a06-4db1-b4b3-ce48280a5c24.jpg) | ![dashboard-after-4](https://user-images.githubusercontent.com/2726132/213233711-30282a14-0bd7-44c1-8878-f70371f27528.jpg)|
